### ISSUE TITLE
main

### ISSUE THREAD

**Issue by ramnerd** [ramnerd/Internship_tasks#2]
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import LabelEncoder
from sklearn.tree import DecisionTreeClassifier, plot_tree
from sklearn.metrics import accuracy_score, classification_report, confusion_matrix
import matplotlib.pyplot as plt

# Load the dataset
df = pd.read_csv("task3\customer_data.csv")  

# View basic info
print(df.head())

# Encode categorical features
categorical_columns = ['Gender', 'Marital_Status', 'Device_Type', 'Location']
label_encoders = {}

for col in categorical_columns:
    le = LabelEncoder()
    df[col] = le.fit_transform(df[col])

    label_encoders[col] = le
print(label_encoders)

# Define features and target
X = df.drop('Purchased', axis=1)
y = df['Purchased']

# Split into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Train Decision Tree Classifier
clf = DecisionTreeClassifier(max_depth=5, random_state=42)
clf.fit(X_train, y_train)

# Make predictions
y_pred = clf.predict(X_test)

# Evaluate the model
print("Accuracy:", accuracy_score(y_test, y_pred))
print("\nClassification Report:\n", classification_report(y_test, y_pred))
print("\nConfusion Matrix:\n", confusion_matrix(y_test, y_pred))

# Visualize the tree
plt.figure(figsize=(20,10))
plot_tree(clf, feature_names=X.columns, class_names=["Not Purchased", "Purchased"], filled=True)
plt.title("Decision Tree for Purchase Prediction")
plt.show()
